src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -53,11 +58,12 @@ public List<String> prepareListOfAllBooks() throws NotFoundInDatabaseException {
         for (int i = 0; i < allBooksInDatabase.size(); i++) {
             results.add(allBooksInDatabase.get(i).toString());
         }
-        return results;
-    }
 
-    private List<BookInfo> getAllBooksFromDatabase() {
-        return bookRepository.findAllBooks();
+        if (results.isEmpty()) {
```

Comment:
Again, this can be done above all the logic about loop and so on. After you getting data from database.